Examples/Java/Sources/Image.java

```diff
@@ -21,30 +20,133 @@
-    public abstract @SerializedName("url") @Nullable String url();
-    public abstract @SerializedName("width") @Nullable Integer width();
-    public static Builder builder() {
-        return new AutoValue_Image.Builder();
-    }
-    abstract Builder toBuilder();
-    public static TypeAdapter<Image> jsonAdapter(Gson gson) {
-        return new AutoValue_Image.GsonTypeAdapter(gson);
-    }
-    @AutoValue.Builder
-    public abstract static class Builder {
-    
+    @SerializedName("height") private @Nullable Integer height;
+    @SerializedName("url") private @Nullable String url;
+    @SerializedName("width") private @Nullable Integer width;
+     static final private int HEIGHT_SET = 1 << 0;
+     static final private int URL_SET = 1 << 1;
+     static final private int WIDTH_SET = 1 << 2;
+     private int _bits = 0;
+    private Image(@Nullable Integer height,
+    @Nullable String url,
+    @Nullable Integer width,
+    int _bits) {
+        this.height = height;
+        this.url = url;
+        this.width = width;
+        this._bits = _bits;
+    }
+    public static Image.Builder builder() {
+        return new Image.Builder();
+    }
+    public Image.Builder toBuilder() {
+        return new Image.Builder(this);
+    }
+    public Image mergeFrom(Image model) {
+        Image.Builder builder = this.toBuilder();
+        builder.mergeFrom(model);
+        return builder.build();
+    }
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Image that = (Image) o;
+        return Objects.equals(this.height, that.height) &&
+        Objects.equals(this.url, that.url) &&
+        Objects.equals(this.width, that.width);
+    }
+    @Override
+    public int hashCode() {
+        return Objects.hash(height,
+        url,
+        width);
+    }
+    public @Nullable Integer getHeight() {
+        return this.height;
+    }
+    public @Nullable String getUrl() {
+        return this.url;
+    }
+    public @Nullable Integer getWidth() {
+        return this.width;
+    }
+    public boolean getHeightIsSet() {
+        return (this._bits & HEIGHT_SET) == HEIGHT_SET;
+    }
+    public boolean getUrlIsSet() {
+        return (this._bits & URL_SET) == URL_SET;
+    }
+    public boolean getWidthIsSet() {
+        return (this._bits & WIDTH_SET) == WIDTH_SET;
+    }
+    public static class Builder {
     
-        public abstract Builder setHeight(@Nullable Integer value);
-        public abstract Builder setUrl(@Nullable String value);
-        public abstract Builder setWidth(@Nullable Integer value);
-        public abstract Image build();
+        @SerializedName("height") private @Nullable Integer height;
+        @SerializedName("url") private @Nullable String url;
+        @SerializedName("width") private @Nullable Integer width;
+         private int _bits = 0;
```

Comment:
strange indent here?

Reply:
whoops, thanks. had some code that wasn't handling variables with zero annotations properly. fixed.